trackpy/preprocessing.py

```diff
@@ -77,10 +77,16 @@ def bandpass(image, lshort, llong, threshold=None):
         boxcar = uniform_filter1d(boxcar, size, axis, **settings)
     gaussian = ifftn(fourier_gaussian(fftn(image), lshort)).real
     result = gaussian - boxcar
-    return np.where(result > threshold, result, 0)
+    signal = result > threshold
+    if not analyze_background:
+        return np.where(signal, result, 0)
+    background = image[~signal]
+    return np.where(signal, result, 0), background.mean(), background.std()
 
 
-def scale_to_gamut(image, original_dtype):
-    max_value = np.iinfo(original_dtype).max
-    scaled = (max_value/image.max()*image.clip(min=0.))
-    return scaled.astype(original_dtype)
+def scale_to_gamut(image, original_dtype, return_scale_factor=False):
+    scale_factor = np.iinfo(original_dtype).max / image.max()
+    scaled = (scale_factor * image.clip(min=0.)).astype(original_dtype)
+    if return_scale_factor:
+        return scaled, scale_factor
```

Comment:
This is really bad practice (to change the return type/number of returns) depending on the input.

I would suggest splitting the `scale_factor` computation in to it's own function which gets called here and can be called where ever it is needed elsewhere in the code or to break the API and always return the scale factor.